mempool/src/interface/mock.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(clippy::unwrap_used)]
+
+use crate::error::Error;
+use crate::tx_accumulator::TransactionAccumulator;
+use crate::MempoolEvent;
+use common::chain::{SignedTransaction, Transaction};
+use common::primitives::Id;
+use std::sync::{Arc, Mutex};
+use subsystem::{CallRequest, ShutdownRequest};
+
+use super::mempool_interface::{MempoolInterface, MempoolSubsystemInterface};
+
+#[derive(Clone)]
+pub struct MempoolInterfaceMock {
+    pub add_transaction_called: Arc<Mutex<bool>>,
+    pub add_transaction_should_error: Arc<Mutex<bool>>,
+    pub get_all_called: Arc<Mutex<bool>>,
+    pub get_all_should_error: Arc<Mutex<bool>>,
+    pub contains_transaction_called: Arc<Mutex<bool>>,
+    pub contains_transaction_should_error: Arc<Mutex<bool>>,
+    pub collect_txs_called: Arc<Mutex<bool>>,
+    pub collect_txs_should_error: Arc<Mutex<bool>>,
+    pub subscribe_to_events_called: Arc<Mutex<bool>>,
+    pub subscribe_to_events_should_error: Arc<Mutex<bool>>,
+    pub run_called: Arc<Mutex<bool>>,
+    pub run_should_error: Arc<Mutex<bool>>,
```

Comment:
Not that it is a big deal but maybe this could be improved:

* Use one `Arc` to guard all the fields (unless you want to keep references to the individual fields independently of life of the struct. It involves adding an intermediate struct though
* Use `AtomicBool` in place of a `Mutex`. I think it also removes the need for some `unwrap`s down below.

Reply:
Oh, `AtomicBool`... of course!

There wasn't any immediate need for the separation, it way just more convenient to copy-paste. I'll change to `AtomicBool` and see how it looks with a single containing `Arc`.

Reply:
I think for a mock object, and the way how I'm using it, a single `Arc` per field is fine for now.

But yes, `AtomicBool` was much nicer.